sdk/servicebus/Microsoft.Azure.ServiceBus.sln

```diff
@@ -3,9 +3,9 @@ Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
 VisualStudioVersion = 15.0.28307.421
 MinimumVisualStudioVersion = 10.0.40219.1
```

Comment:
Are we intending to combine the management and client libraries into a single solution?    My preference would be to keep them separate;  I don't see them as one unit of release or work.    Given that the Service Bus client library has a `Management` area that is separate and distinct from the management plane, this could be a source of confusion.

Reply:
I think we should keep the solution to just the projects where there is manual coding and since the management libraries are completely auto-generated I would keep them out of these solutions.

Reply:
I'm 100% in favor.    That would mean we need to split the root directory, right?   In your guidance, I think that would look something like:

```
/sdk/service-bus/
|----Microsoft.Azure.ServiceBus
       |---- << Client Solution File >>|
|----Microsoft.Azure.Management.ServiceBus
       |---- << Managment Solution File >>
```

Right?

Reply:
Solution files should be scoped to a parent folder which has all the child projects under it. So in the case of ServiceBus there is only the one package and I think the solution should go as you diagram it under the Microsoft.Azure.ServiceBus directory. However for some libraries there might be a solution file in the root directory that references multiple of the library projects like keyvault.

Reply:
Makes sense to me.   The thing that I was making sure that there was alignment on was having isolation between client and management, with each in their own solutions.

Reply:
@chidozieononiwu it looks like you still need to move this sln file into the Microsoft.Azure.ServiceBus directory.

Reply:
Done

Reply:
I think that we're still off a level, if we're considering standard .NET conventions and if I understand the new structure.   I believe that we're missing a project container under `src`  and one under `tests`.  Normally these would have the name of the project, though we could (and I'd prefer to) shorten the name to `src/ServiceBus` and `tests/ServiceBus`  in the name of keeping paths shorter and avoiding redundancy.

What we have now is basically:
```
/sdk/service
|----Top Level Package
     |----Solution
     |----src
          |---- Files for every project that is not test-related in the solution
     |----tests
          |---- files for every project that is test-related in the solution
````

What I would expect to see as a .NET developer:
```
/sdk/service
|----Top Level Package
     |----Solution
     |----src
          |----ServiceBus
              |---- Project file and related assets
     |----tests
          |----ServiceBus
              |----Project file and related assets
````

For Service Bus, where there is a single `src` project and a single `test` project, what we have is clean.   However, if we consider something like Event Hubs which has three projects that would live under `src`,  it's messy, would be confusing, and has a non-trivial chance of having name collisions between files.   

In our current structure, Event Hubs would look like:
```
/sdk/eventhub/
|----Microsoft.Azure.EventHubs
       |----Solution
       |----src
            |----EventHubs project file and related assets
            |----EventHubs.Processor project file and related assets
            |----EventHubs.ServiceFabric.Processor project file and related assets
       |----tests
            |---- Test project file and related assets
```

As a .NET developer, I would expect to see something closer to:
```
/sdk/eventhub/
|----Microsoft.Azure.EventHubs
       |----Solution
       |----src
             |----EventHubs
                   |---- EventHubs project file and related assets
             |----EventHubs.Processor
                   |----EventHubs.Processor project file and related assets
             |----EventHubs.ServiceFabric.Processor
                   |----EventHubs.ServiceFabric.Processor project file and related assets
       |----tests
            |----EventHubs.Tests
                 |---- Test project file and related assets
```
_(Note that, for better or worse, Event Hubs has one test project for all of the projects in the solution)_


@weshaggard:  I don't mean to derail, but would you be so kind as to clarify the intent?

Reply:
@jsquire for EventHubs I assume each of those projects are a separate shipping package correct? If so I think the structure would be more like:
```
/sdk/eventhub/
|----Solution - while this isn't ideal because it doesn't contain all the project I think it is OK as long as it is named appropriately. 
|----Microsoft.Azure.EventHubs
       |----src
       |----tests
|----Microsoft.Azure.EventHubs.Processor
       |----src
       |----tests
|----Microsoft.Azure.EventHubs.ServiceFabric.Processor
       |----src
       |----tests
```
Ideally event hubs would have one test project per package and if it doesn't for now it can simply use the one test project and reference the other projects using relative paths. This is a similar pattern that we did for keyvault as well.

Reply:
Where would you expect to see common shared infrastructure in that model?   If you're looking at a single service with multiple packages, they'll invariably share a common baseline.   Normally, this would be in something like `Service.Core` or `Tests.Common` or whatever nomenclature is used.    That doesn't cleanly align to the grouping everything by package.

Reply:
I would prefer not to use a fake package but instead put the common shared things in the core base most package directory. So Microsoft.Azure.EventHubs in this case. For example keyvault has a common TestFramework library that we put here https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/keyvault/Microsoft.Azure.KeyVault/tests/TestFramework. While it isn't ideal it gets the job done. 

The reason I'd like to avoid fake packages is because we will have tooling in place that will use our structure to enforce things like a README and changelog will exist for every package and the more fake packages we have the more exclusions and baselines we need for that tooling.